interface/main/messages/templates/linked_documents.php

```diff
@@ -0,0 +1,237 @@
+        $docInformation['hasPatient'] = true;
+        $docInformation['pname'] = $prow['fname'] . " " . $prow['lname'];
+        $docInformation['pid'] = $prow['pid'];
+        $docInformation['pubpid'] = $prow['pubpid'];
+        $docInformation['DOB'] = $prow['DOB'];
+    }
+    if ($cdaDocumentValidator->isCdaDocument($d)) {
+        $docInformation['isCda'] = true;
+        $docInformation['requiresValidation'] = true;
+    }
+    $records[] = $docInformation;
+}
+$twig = new TwigContainer(null, $GLOBALS['kernel']);
+try {
+    foreach ($records as $record) : ?>
+<div class="row mt-2 mb-2 messages-document-row <?php echo $record['requiresValidation'] ? "messages-document-validate" : ""; ?>"
+     data-doc="<?php echo $record['documentId']; ?>">
+    <div class="col-12">
+    <span class='font-weight-bold'><?php echo xlt('Linked document'); ?>:</span>
+        <?php if (!$record['hasPatient']) : ?>
+        <a class='messages-document-link<?php echo $record['requiresValidation'] ? " d-none" : ""; ?>'
+           href='javascript:void(0);' onClick="previewDocument(<?php echo attr_js($record['documentId']); ?>);"
+
+        >
+            <?php echo text($record['title']); ?>
+        </a>
+        <?php else : ?>
+            <a class='messages-document-link <?php echo $record['requiresValidation'] ? "d-none" : ""; ?>'
+               href='javascript:void(0);'
+               onClick="gotoReport(<?php echo attr_js($record['documentId']); ?>, '<?php echo attr_js($record['pname']); ?>', '<?php echo attr_js($record['pid']); ?>','<?php echo attr_js($record['pubpid'] ?? $record['pid']); ?>','<?php echo attr_js($record['DOB']); ?>');">
+                <?php echo text($record['title']); ?>
+            </a>
+        <?php endif; ?>
+        <?php if ($record['requiresValidation']) : ?>
+            <span class="validation-line">
+                <?php echo text($record['title']); ?>
+                <span class="text-info"><?php echo xlt("Validating document for errors"); ?>...</span>
+                <span class="spinner-border-sm spinner-border" role="status">
+                    <span class="sr-only"><?php echo xlt("Validating document for errors"); ?>...</span>
+                </span>
+            </span>
+            <span class="validation-failed text-danger d-none">
+                <?php echo text($record['title']); ?> -
+                <?php echo xlt("Failed to retrieve validation results from server"); ?>
+            </span>
+        <span class="validation-totals validation-totals-success badge badge-pill badge-success p-2 d-none">
+            <?php echo xlt("Errors"); ?> <span class="validation-totals-count">0</span>
+        </span>
+            <span class="validation-totals validation-totals-failed badge badge-pill badge-danger p-2 d-none">
+            <?php echo xlt("Errors"); ?> <span class="validation-totals-count">0</span>
+        </span>
+        <?php endif; ?>
+        <?php if ($record['isCda']) : ?>
```

Comment:
to avoid php8 issues, generally use !empty() or empty when checking for existence of something in a conditional

Reply:
isCda is a boolean true / false, it won't ever be empty.

Reply:
fixed.